connectivity/cellular/source/framework/RIL/RIL_CellularContext.cpp

```diff
@@ -0,0 +1,1164 @@
+}
+
+void RIL_CellularContext::call_network_cb(nsapi_connection_status_t status)
+{
+    if (_connect_status != status) {
+        _connect_status = status;
+        if (_status_cb) {
+            _status_cb(NSAPI_EVENT_CONNECTION_STATUS_CHANGE, _connect_status);
+        }
+        if (_nw && _connect_status == NSAPI_STATUS_DISCONNECTED) {
+            tr_info("CellularContext disconnected with cid: %d and apn: %s", _cid, _apn);
+            disconnect(false); // Clean up network interface objects if network dropped
+        }
+    }
+}
+
+void RIL_CellularContext::stack_status_cb(nsapi_event_t ev, intptr_t ptr)
+{
+    tr_debug("RIL_CellularContext::stack_status_cb: event %d, ptr %d", ev, ptr);
+
+    if (ev == NSAPI_EVENT_CONNECTION_STATUS_CHANGE && ptr == NSAPI_STATUS_GLOBAL_UP) {
+        _is_connected = true;
+    } else {
+        _is_connected = false;
+    }
+
+    // call device's callback, it will broadcast this to here (cellular_callback)
+    _device->cellular_callback(ev, ptr, this);
+}
+
+nsapi_connection_status_t RIL_CellularContext::get_connection_status() const
+{
+    if (_stack_interface) {
+        return _stack_interface->get_connection_status();
+    } else {
+        return _connect_status;
+    }
+}
+
+void RIL_CellularContext::request_deactivate_data_call_response(ril_token_t *token, RIL_Errno err, void *response,
+                                                                size_t response_len)
+{
+    if (err != RIL_E_CANCELLED) {
+        token->cond_mutex->lock();
+        token->response_error = (err == RIL_E_SUCCESS) ? NSAPI_ERROR_OK : NSAPI_ERROR_DEVICE_ERROR;
+        token->cond_var->notify_one();
+        token->cond_mutex->unlock();
+    }
+}
+
+void RIL_CellularContext::unsolicited_response(int response_id, const void *data, size_t data_len)
+{
+    tr_debug("RIL_CellularContext::unsolicited_response: id = %d", response_id);
+
+    if (response_id == RIL_UNSOL_DATA_CALL_LIST_CHANGED) {
+        const int items = data_len / sizeof(RIL_Data_Call_Response_v11);
+        MBED_ASSERT(data_len == items * sizeof(RIL_Data_Call_Response_v11));
+        if (items) {
+            MBED_ASSERT(data);
+            const RIL_Data_Call_Response_v11 *resp_data = (const RIL_Data_Call_Response_v11 *)data;
+            for (int i = 0; i < items; i++) {
+                const RIL_Data_Call_Response_v11 &resp_item = resp_data[i];
+                if (resp_item.cid == _cid) { // this context cid, continue processing
+                    if (resp_item.active == 0 && resp_item.status != PDP_FAIL_NONE && resp_item.suggestedRetryTime == -1) {
+                        // This context was active, clean up the network interface and send NSAPI_STATUS_DISCONNECTED to application
+                        call_network_cb(NSAPI_STATUS_DISCONNECTED);
+                    }
+                    return;
+                }
+            }
+        }
+    }
+}
+
+const char *RIL_CellularContext::pdp_type_to_string(pdp_type_t pdp_type)
```

Comment:
Remove this, as it should be already in `CellularContext`.

Reply:
This PR is based on mbed os 6 to add RIL interface, the pr-S5JS100-06_RIL is based on mbed os 5, so it not in `CellularContext`

Reply:
Again the same thing. This has not needed earlier, but as it's counterpart (`string_to_pdp_type`) is in `CellularContext`, this should be added there as well.

Reply:
OK, I got it and will to add it, thanks